source/leuven/slurm_specifics.rst

```diff
@@ -173,3 +173,98 @@ A few notes on this feature:
+   :widths: 20 20 20 20 20
+   :header-rows: 2
+
+   * - Cluster
+     - Partition(s)
+     - CPU cores
+     - ``--mem-per-cpu``
+     - ``--mem``
+   * -
+     -
+     -
+     - (MiB)
+     - (MiB)
+   * - Genius
+     - ``gpu_p100*``
+     - 9
+     - 5000
+     - 45000
+   * - Genius
+     - ``gpu_v100*``
+     - 4
+     - 21000
+     - 84000
+   * - wICE
+     - ``interactive``
+     - 8
+     - 7500
+     - 60000
+   * - wICE
+     - ``gpu|gpu_a100``
+     - 18
+     - 7000
+     - 126000
+   * - wICE
+     - ``gpu_h100``
+     - 16
+     - 11700
+     - 187200
+
+The following remarks apply when submitting jobs to the GPU partitions:
+
+* If a job requests more cores or memory per every GPU as listed above, the job
+  will not be submitted to the queue.
+  Instead, an informative message will be sent to the standard error stream.
+* A job may request less cores and/or memory per GPU than the maximum limit in the table above.
+* Instead of specifying ``--mem`` or ``--mem-per-cpu``, a job may request ``mem-per-gpu``.
+  In that case, the maximum value specified for ``--mem`` applies to ``--mem-per-gpu``, too.
+* For restricting maximum memory for GPU jobs, one has to choose one of the ``--mem``, ``--mem-per-cpu``,
+  or ``--mem-per-gpu`` options.
+* For multi-GPU jobs, the multiple of resouces from the table above applies.
+  E.g. the maximum allowed resources for a two-GPU job on wICE ``gpu_a100`` partition would look like:
+
+  .. code-block:: bash
+
+     sbatch --account=lp_myproject --clusters=wice --partition=gpu_a100 \
+            --nodes=1 --ntasks=36 --gpus-per-node=2 --mem=252000m \
+            myjobscript.slurm
+
+  Similarly, multi-node multi-GPU jobs can take up the entire cores and memory of the nodes.
+  But, resources can be specified per node and device:
+
+
+  .. code-block:: bash
+
+     sbatch --account=lp_myproject --clusters=wice --partition=gpu_a100 \
+            --nodes=2 --ntasks-per-gpu=18 --gpus-per-node=4 --mem-per-gpu=126000m \
+            myjobscript.slurm
+
+* Due to the Multi-Instance GPU (MIG) configuration of the Nvidia A100 GPUs on the
+  wICE ``interactive`` partition, specifying ``--gpus-per-node=1`` will result in
+  allocation of 1/7th of the physical device.
+  One cannot request any additional GPU instance from this partition.
+* Slurm supports `GPU sharding <https://slurm.schedmd.com/gres.html#Sharding>`_, and this
+  feature is enabled for all our GPUs.
+  The maximum shards per each GPU is equivalent to the number of cores of the compute host.
+  When requesting GPU shards, *no* resource limits apply.
+  In this case, the user is supposed to request the same number of cores as the requested
+  GPU shards.
+  It is also adviced to leave out memory specifications, and rely on the default memory per core.
+* All the examples given in the :ref:`Genius <genius_t2_leuven>` and :ref:`wICE <wice_t2_leuven>`
+  quick start guides fully comply with the correct resource proportions.
```

Comment:
- For now there's only a warning; non-compliant jobs are not (yet) rejected.
- From the above change it should already be evident that these are upper bounds (and so that jobs remain free to request less resources per GPU).
- We should keep in mind that users will only need to bother with these constraints if they've intentionally raised the CPU resources for their job (the default is one core per node and default memory per core, so that always complies). As a result, one example should be enough (at least for now).
- I don't see the added value of the interactive partition information in this context (and it's of course already covered in https://docs.vscentrum.be/leuven/wice_quick_start.html#submit-to-the-interactive-partition).
- Shards are a resource we only support in the context of trainings where it is needed to accommodate many users on a limited number of devices. Regular jobs/users are not supposed to use this feature and (IMO) should not even be made aware of it.

So based on this my suggestion is:

```suggestion
The submit filter will issue a warning if a job requests more cores or memory per GPU
than what is listed above. If this happens, please adjust the Slurm options accordingly
for your future jobs.

As an example, suppose that you need two A100 GPUs for your calculation, with just
one core per GPU but with as much CPU memory as you can get. Such a job can be
submitted as follows:

  .. code-block:: bash
     # This job will get less than 18 cores per GPU, so this requirement is satisfied
     # It will receive 126000 MiB of CPU memory per GPU, which is the maximum
     # we can get without getting the submit filter warning
     sbatch --account=lp_myproject --clusters=wice --partition=gpu_a100 \
            --nodes=1 --ntasks-per-node=2 --gpus-per-node=2 --mem=252000 \
            myjobscript.slurm

For more examples of valid GPU jobs, have a look at the
:ref:`Genius <genius_t2_leuven>` and :ref:`wICE <wice_t2_leuven>`
quickstart guides.

Aside from options such as ``--ntasks-per-node`` and ``--cpus-per-task``
(for CPU cores) and ``--mem`` and ``--mem-per-cpu`` (for CPU memory),
keep in mind that Slurm also offers options like ``--cpus-per-gpu`` and
``--mem-per-gpu``.
```